### DISCUSSION TITLE
Reference Shell implementations

### DISCUSSION THREAD

**Discussion by polarhive** [homebrew-ec-foss/psh#3] [Reading and Resources]
https://brennan.io/2015/01/16/write-a-shell-in-c
https://github.com/brenns10/lsh/blob/master/src/main.c

ref: #1

**Comment by tejas-techstack** [comment]
https://github.com/danishprakash/dash
another easy shell implementation

**Comment by g-anupam** [comment]
https://www.geeksforgeeks.org/introduction-linux-shell-shell-scripting/ - What exactly is a shell?

**Comment by ninsid711** [comment]
[https://youtu.be/IYZDIhfAUM0?si=_hnUtAWekIPorp4n](url)
just gives a proper run down of some shell commands, felt like this might be helpful
